File: autumn_project_frontend/ferret/src/urls.js
// Backend URls

export const OAUTH_AUTH_ROOT = 'https://channeli.in/oauth/authorise/'
export const CLIENT_ID = 'AIEwG4pJKHWdgNWHeaUs56KpDNfHyEgBQLKRz1DL'
export const REDIRECT_URI = 'http://localhost:8000/auth/login/'
export const STATE = 'foo_success216'
export const OAUTH_AUTH = `${OAUTH_AUTH_ROOT}?response_type=code&client_id=${CLIENT_ID}&redirect_uri=${REDIRECT_URI}&state=${STATE}`

export const BACKEND_BASE_URL = 'http://localhost:8000/'
export const IS_USER_AUTHENTICATED = `${BACKEND_BASE_URL}api/is_user_authenticated/`
export const USERS = `${BACKEND_BASE_URL}user/`
export const SEASONS_BY_TYPE = `${BACKEND_BASE_URL}recruitment_season/`
export const ROUNDS = `${BACKEND_BASE_URL}round/`
export const SECTIONS = `${BACKEND_BASE_URL}section/`
export const QUESTIONS = `${BACKEND_BASE_URL}question/`
export const CANDIDATES = `${BACKEND_BASE_URL}candidate/`
export const CANDIDATE_ROUND = `${BACKEND_BASE_URL}candidate_round/`
export const CANDIDATE_MARKS = `${BACKEND_BASE_URL}candidate_marks/`
export const SECTION_MARKS = `${BACKEND_BASE_URL}api/section_marks/`
export const CANDIDATE_SECTION_MARKS = `${BACKEND_BASE_URL}api/candidate_section_marks/`
export const LOGOUT = `${BACKEND_BASE_URL}auth/logout/`
export const CSV = `${BACKEND_BASE_URL}api/csv/`
export const FILTER_CANDIDATES = `${BACKEND_BASE_URL}api/filter_candidates/`
export const INTERVIEW_PANEL = `${BACKEND_BASE_URL}interview_panel/`


// Frontend URLs

export const FRONTEND_BASE_URL = 'http://localhost:3000/'
export const HOME = `/home`
export const SEASON_DASHBOARD = `/season`

File: autumn_project/recruiter/urls.py
from django.urls import path,include
from .views import *
from rest_framework.routers import DefaultRouter,SimpleRouter

router = DefaultRouter()

router.register('recruitment_season',RecruitmentSeasonsModelViewSet ,basename='season')
router.register('round',RoundsModelViewSet ,basename='round')
router.register('section',SectionsModelViewSet ,basename='section')
router.register('question',QuestionsModelViewSet ,basename='question')
router.register('user',UsersModelViewSet ,basename='user')
router.register('candidate',CandidatesModelViewSet ,basename='candidate')
router.register('interview_panel',InterviewPanelModelViewSet ,basename='interview_panel')
router.register('candidate_round',CandidateRoundModelViewSet ,basename='candidate_round')
router.register('candidate_marks',CandidateMarksModelViewSet ,basename='candidate_marks')
router.register('candidate_link',CandidateProjectLinkModelViewSet ,basename='candidate_link')

urlpatterns =[
    path('', include(router.urls)),
    path('api-auth/', include('rest_framework.urls')),
    path('auth/auth_code/', getAuthCode.as_view()),
    path('auth/login/', LoginView.as_view()),
    path('api/is_user_authenticated/', isUserAuthenticated.as_view()),
    path('auth/logout/', LogoutView.as_view()),
    path('api/csv/', UploadCSV.as_view()),
    path('api/section_marks/', SectionMarksView.as_view()),
    path('api/candidate_section_marks/', CandidateSectionMarks.as_view()),
    path('api/filter_candidates/', FilterCandidatesView.as_view())
]